varats-core/varats/utils/git_util.py

```diff
@@ -988,3 +989,37 @@ def branch_has_upstream(
             branch_name + "@{upstream}"] | grep[upstream]
     ) & RETCODE
     return tp.cast(bool, exit_code == 0)
+
+
+def calc_surviving_lines(repo: pygit2.Repository, revision) -> dict[str, int]:
```

Comment:
```suggestion
def calc_surviving_lines(repo: pygit2.Repository, revision) -> tp.Dict[str, int]:
```
Only newer python version support direct use of types